aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1143,8 +1139,8 @@ impl AptosVM {
         // TODO(Gas): revisit this.
         init_speculative_logs(1);
 
-        let storage = StorageAdapter::new(state_view);
-        let vm = AptosVM::new(&storage);
+        let resolver = StorageAdapter::new(state_view);
```

Comment:
I actually don't remember why we need this StorageAdapter, can't we just impl<T: StateView> MoveResolverExt for T?

Reply:
Unfortunately not according to rust language rules

Reply:
But yes that is exactly what it should be, and it also indicates that the owning here is wrong.